### PR TITLE
TheMealDB (Independent Publisher)

### PR DESCRIPTION
**Pull request by johnmuchiriM**
independent-publisher-connector
New
student
proposal

---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [x] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by johnmuchiriM** [comment]
RandomMeal
SearchForMealWithId
Ingredients

**Comment by johnmuchiriM** [comment]
ValidatedSuccefully

**Comment by johnmuchiriM** [comment]
Hello, @sriyen-msft I am requesting a review of The MealDBConnector. 
Regards,
John.

File: independent-publisher-connectors/The Meal DB/TheMealDB.swagger.json [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    "/lookup.php": {
+      "get": {
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
+          }
+        },
+        "summary": "Look for a Meal with id",
+        "description": "Lookup full meal details by id",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/The Meal DB/TheMealDB.swagger.json:47]: @johnmuchiriM, all operations descriptions should end with a period, No period required only if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/The Meal DB/TheMealDB.swagger.json [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    "/random.php": {
+      "get": {
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
+          }
+        },
+        "summary": "Get a random meal",
+        "description": "Lookup a random meal",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/The Meal DB/TheMealDB.swagger.json:69]: @johnmuchiriM, all operations descriptions should end with a period, No period required only if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/The Meal DB/TheMealDB.swagger.json [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    "/list.php": {
+      "get": {
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
+          }
+        },
+        "summary": "List all Areas",
+        "description": "List all Areas where meals are mostly taken",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/The Meal DB/TheMealDB.swagger.json:84]: @johnmuchiriM, all operations descriptions should end with a period, No period required only if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/The Meal DB/TheMealDB.swagger.json [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    "/filter.php": {
+      "get": {
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
+          }
+        },
+        "summary": "Filter meals by ingredient",
+        "description": "Filter by main ingredient",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/The Meal DB/TheMealDB.swagger.json:106]: @johnmuchiriM, all operations descriptions should end with a period, No period required only if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/The Meal DB/TheMealDB.swagger.json [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    "/categories.php": {
+      "get": {
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
+          }
+        },
+        "summary": "List all meals categories",
+        "description": "List all meal categories",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/The Meal DB/TheMealDB.swagger.json:128]: @johnmuchiriM, all operations descriptions should end with a period, No period required only if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by johnmuchiriM** [comment]
Hello @Amjed-Ayoub, 
I made the changes requested in the conversation. I am requesting a review of the submission.
Regards,
John

**Comment by Amjed-Ayoub** [comment]
Hi @johnmuchiriM, Thank you so much for all your hard work with us and thank you for fixing your connector artifacts.
May I ask you please to resolve all conversations above after you make the changes.

**Comment by johnmuchiriM** [comment]
Hi @Amjed-Ayoub, I have resolved all the conversations. Welcome & thanks for the nice words.

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @johnmuchiriM,
I hope you are doing well. 
May I ask you please to remove the other proposal files from this pull request so we can move forward with the certification process.
Please let me know if you have any questions.
Thank you so much again.

**Comment by johnmuchiriM** [comment]
Hi, I have deleted a readme file for another separate connector. Kindly @Amjed-Ayoub advise if we're all good.

**Comment by Amjed-Ayoub** [comment]
> Hi @Amjed-Ayoub, I am doing well, I hope it is the same on your end. I have deleted a readme file for another separate connector. Kindly advise if we're all good. I also have one question on **needs: partner-attention** badge, does it mean asking permission from the API owner? I have attached a request I made to the API owner, Is it the one? kindly expand on it. Thank you & welcome. PartnerInfo

Hi @johnmuchiriM Thank you so much for that question. this label is for you as a publisher to know that the pull request needs your attention to fix something. for example here it was to let you know that you needed to delete the readme file for the other proposal.
I hope this helps.
Thank you so much again.

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @johnmuchiriM,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by johnmuchiriM** [comment]
Hi @Amjed-Ayoub,
Thank you so much for the in-depth explanation. I have a clear understanding of the badge. The next time I see it, it will not catch me off-guard.

**Comment by johnmuchiriM** [comment]
Hi @Amjed-Ayoub,
Great news! I am excited to get to this level. I don't have a question at the moment. 🔥

**Comment by Amjed-Ayoub** [comment]
Hello @johnmuchiriM ,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by johnmuchiriM** [comment]
Hi @Amjed-Ayoub ,

These are great news! I am excited to see it go to the last stages. I will keep calm and await its deployment.

Best regards,
John Muchiri.

**Comment by Amjed-Ayoub** [comment]
> Hi @Amjed-Ayoub ,
> 
> These are great news! I am excited to see it go to the last stages. I will keep calm and await its deployment.
> 
> Best regards, John Muchiri.

Hello @johnmuchiriM,
Thank you so much for all your hard work on this connector. We really appreciate your enthusiasm and excitement.
We are expediting the deployment process for all new connectors. Please expect this to complete within this week :-)

Thank you so much again for working with us.

**Comment by johnmuchiriM** [comment]
Hi @Amjed-Ayoub ,
This is well noted. The whole journey is exciting and it makes it hard to hide. Thank you for your constant follow-up and communication.
Best regards,
John Muchiri

**Comment by Amjed-Ayoub** [comment]
Hello @johnmuchiriM,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Review by Amjed-Ayoub — APPROVED** [review]

**Comment by johnmuchiriM** [comment]
Hi @Amjed-Ayoub 
These are just the news I needed to end the week in an awesome way! I am so happy! Thank you for the guidance you offered me through the whole process. This deployment journey has been awesome, I loved it! I will check on the survey. Looking forward on the connector being used by platform developers and other users! This is awesome!!!

**Comment by Amjed-Ayoub** [comment]
Thank you so much @johnmuchiriM, it was a pleasure working with you.
We are looking forward to your next connector.
Thank you so much again.

**Comment by johnmuchiriM** [comment]
Hi @Amjed-Ayoub ,
You're Welcome. The next connector is on the way!

**Comment by Amjed-Ayoub** [comment]
Hi @johnmuchiriM,

I hope you are doing well.

I just noticed that I mistakenly approved your PR as it is missing the contact section in the swagger file. May I ask you please to add this section below to your swagger file and submit a new PR. We will move fast with the deployment since this is a very small change. This will allow us to send you your Biweekly report about your connector usage.
I look forward to hearing from you.
Thank you so much.
image

**Comment by johnmuchiriM** [comment]
Hi @Amjed-Ayoub ,
This is well noted. I will do the changes and inform you once I push to a new PR. 
Thanks.

**Comment by johnmuchiriM** [comment]
Hi @Amjed-Ayoub ,
I created a new PR. Named it The Meal DB v1.1.0 
Thank you.
John.

**Pull request merged**